AirCastingTests/API/SyncContextServiceTests.swift

```diff
@@ -0,0 +1,171 @@
+// Created by Lunar on 19/06/2021.
+//
+
+import XCTest
+import Combine
+@testable import AirCasting
+
+class SyncContextServiceTests: XCTestCase {
+    let validContextResponseData: Data =
+        """
+        {
+            "download": ["UUID-TO-DOWNLOAD1", "UUID-TO-DOWNLOAD2"],
+            "upload": ["UUID-TO-UPLOAD1", "UUID-TO-UPLOAD2"],
+            "deleted": ["UUID-TO-DELETE1", "UUID-TO-DELETE2"],
+        }
+        """.data(using: .utf8)!
+    let client = APIClientMock()
+    let auth = RequestAuthorizationServiceMock()
+    let responseValidator = HTTPResponseValidatorMock()
+    lazy var service = SessionSynchronizationService(client: client, authorization: auth, responseValidator: responseValidator)
+    private var cancellables: [AnyCancellable] = []
+    
+    func test_callsEndponitCorrectly() {
+        setupWithCorrectDataReturned()
+        let exp = expectation(description: "Will call correct endpoint")
+        service
+            .getSynchronizationContext(localSessions: .random)
+            .sink(receiveCompletion: { _ in }, receiveValue: { _ in
+                XCTAssertEqual(self.client.callHistory.count, 1)
+                let request = self.client.callHistory.first!
+                XCTAssertEqual(request.url?.absoluteString, "http://aircasting.org/api/user/sessions/sync_with_versioning.json")
+                XCTAssertEqual(request.httpMethod, "POST")
+                XCTAssertEqual(request.allHTTPHeaderFields?["Accept"], "application/json")
+                XCTAssertEqual(request.allHTTPHeaderFields?["Content-Type"], "application/json")
+                exp.fulfill()
+            })
+            .store(in: &cancellables)
+        wait(for: [exp], timeout: 0.2)
```

Comment:
This is probably isn't the most readable code and the entry could be hard for less experienced. maybe we can for with completions for the api level ?

Reply:
Also please be advise that this is changing some of conventions that we have accepted. Maybe a good idea it to confront it with others

Reply:
> Also please be advise that this is changing some of conventions that we have accepted. Maybe a good idea it to confront it with others

would be easier for me if you mentioned what are the exact conventions

Reply:
> This is probably isn't the most readable code and the entry could be hard for less experienced. maybe we can for with completions for the api level ?

I'll create a generic helper for spying on observables to make those tests clean. I think it's not worth cluttering production code in order to make tests more readable, but luckily we can have both here.